common/src/parsing.rs

```diff
@@ -1,5 +1,5 @@
 //! Parsing utilities.
```

Comment:
I don't think we should be writing a parsing library like this when [`nom`](https://docs.rs/nom) or [`winnow`](https://docs.rs/winnow) exist - happy to walk you through usage of them

Reply:
I am reusing existing zero-bin parsing implementation. Feel free to open github issue to refactor this code to use `nom` or `window`.

Reply:
#99